platform/golden/bullseye.pkr.hcl

```diff
@@ -1,59 +1,107 @@
+}
+
+variable "ssh_password" {
+  type    = string
+  default = "packer"
+}
+
+variable "ssh_username" {
+  type    = string
+  default = "packer"
 }
 
 source "virtualbox-iso" "base-debian-amd64" {
   boot_command         = [
-	"<esc><wait>",
-        "install <wait>",
-        " auto=true",
-        " priority=critical",
-        " preseed/url=http://{{ .HTTPIP }}:{{ .HTTPPort }}/preseed2.cfg PACKER_USER=packer PACKER_AUTHORIZED_KEY={{ .SSHPublicKey | urlquery }}<enter>",
-	"debian-installer=en_US.UTF-8 <wait>",
-        "auto <wait>",
-        "locale=en_US.UTF-8 <wait>",
-        "kbd-chooser/method=us <wait>",
-        "keyboard-configuration/xkb-keymap=us <wait>",
-        "netcfg/get_hostname={{ .Name }} <wait>",
-        "netcfg/get_domain=vagrantup.com <wait>",
-        "fb=false <wait>",
-        "debconf/frontend=noninteractive <wait>",
-        "console-setup/ask_detect=false <wait>",
-        "console-keymaps-at/keymap=us <wait>",
-        "grub-installer/bootdev=/dev/sda <wait>",
-	"<enter><wait>"
+    "<esc><wait>",
+    "auto <wait>",
+    "console-keymaps-at/keymap=us <wait>",
+    "console-setup/ask_detect=false <wait>",
+    "debconf/frontend=noninteractive <wait>",
+    "debian-installer=en_US <wait>",
+    "fb=false <wait>",
+    "install <wait>", 
+    "kbd-chooser/method=us <wait>",
+    "keyboard-configuration/xkb-keymap=us <wait>",
+    "locale=en_US <wait>",
+    "netcfg/get_hostname=debian-11-7 <wait>",
+    "preseed/url=http://{{ .HTTPIP }}:{{ .HTTPPort }}/${var.config_file} <wait>",
+     "<enter><wait>"
   ]
-
-  http_directory       = "cfg"
-  cpus                 = 8
-  disk_size            = 65535
   guest_os_type        = "Debian11_64"
-  guest_additions_mode = "disable"
-  guest_additions_interface = "virtio"
-  headless             = true
+  cpus                 = "${var.cpus}"
+  disk_size            = "${var.disk_size}"
+  headless             = "${var.headless}"
   output_directory     = "build"
-  output_filename      = "golden.raw"
+  output_filename      = "${var.image_name}.raw"
+  http_directory       = "${var.config_dir}"
   communicator         = "ssh"
   skip_nat_mapping     = false
   iso_checksum         = "${var.iso_checksum_type}:${var.iso_checksum}"
-  iso_interface        = "virtio"
   iso_url              = "${var.iso_url}"
-  memory               = 16384
+  iso_interface        = "virtio"
+  memory               = "${var.memory}"
   nic_type             = "virtio"
   hard_drive_interface = "virtio"
   rtc_time_base        = "UTC"
-  shutdown_command     = "echo 'packer' | sudo -S shutdown -P now"
-  ssh_username         = "packer"
-  ssh_password         = "packer"
-  ssh_wait_timeout     = "5m"
+  shutdown_command     = "echo '${var.ssh_username}' | sudo -S shutdown -P now"
+  ssh_username         = "${var.ssh_username}"
+  ssh_password         = "${var.ssh_password}"
+  ssh_wait_timeout     = "30m"
+  vboxmanage = [
+    [ "modifyvm", "{{.Name}}", "--nat-localhostreachable1", "on" ]
```

Comment:
I am surprised this is enough. I would rather not require a nat network at all for virtualbox, and perhaps this is the way to do that.

One of us has to integrate the other's work. Up to you, you decide :)

Reply:
I'm not sure what the best approach here is for this, the NAT requirement appears to be baseline in VirtualBox, it is used (in reverse) to pull the preseed file via http, and this particular command was needed to allow the preseed to be loaded.

The inverse (e.g. ssh login) appears to also be built in to the packer model (there are comments in the instructions that indicate that if you don't define the ssh connector it is still what is used, and packer will specifically generate the needed ssh user/password/secrets/etc. in order to connect to the host).

In this build, the only thing that happens with SSH is that it is effectively used to determine when the build is complete.  And it is used as a way to run the `shutdown_command` in the boot script as far as I can tell _after_ the system has rebooted from the initial preseed scripted install.